samples/bluetooth/central_otc/src/main.c

```diff
@@ -0,0 +1,522 @@
+/** @file
+ *  @brief Bluetooth Object Transfer Client Sample
+ *
+ * Copyright (c) 2020-2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <sys/printk.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/gatt_dm.h>
+
+#include <sys/byteorder.h>
+#include <bluetooth/scan.h>
+#include <dk_buttons_and_leds.h>
+#include <bluetooth/services/ots.h>
+
+#define BT_UUID_SERVICE_TO_SCAN BT_UUID_OTS
+#define BT_OTS_NAME_MAX_SIZE 127
+#define FIRST_HANDLE			0x0001
+#define LAST_HANDLE			0xFFFF
+#define OBJ_MAX_SIZE 102400
+#define	BT_GATT_OTS_OLCP_RES_OUT_OF_BONDS 0x05
+
+struct bt_ots_client otc;
+static struct bt_ots_client_cb otc_cb;
+unsigned char bwData[OBJ_MAX_SIZE] = {0};
+bool first_selected = false;
+void on_obj_selected(struct bt_ots_client *ots_inst,
+			     struct bt_conn *conn, int err);
+
+void on_obj_metadata_read(struct bt_ots_client *ots_inst,
+				  struct bt_conn *conn, int err,
+				  uint8_t metadata_read);
+                  
+int on_obj_data_read(struct bt_ots_client *ots_inst,
+			     struct bt_conn *conn, uint32_t offset,
+			     uint32_t len, uint8_t *data_p, bool is_complete);
+
+NET_BUF_SIMPLE_DEFINE_STATIC(otc_obj_buf, CONFIG_BT_BUF_ACL_RX_SIZE*4);

```

Comment:
This should be related to BT_OTS_L2CAP_CHAN_RX_MTU which is assigned as value of CONFIG_BT_BUF_ACL_RX_SIZE in zephyr\subsys\bluetooth\services\ots\Kconfig.
You are right, ir should be changed to meaningful definition.
While object reading with bulk data , it is very easy to trigger net_buf_simple_tailroom ASSERT.
So 4 times of BT_OTS_L2CAP_CHAN_RX_MTU is a safer buffer size.

Reply:
But where does the value `4` come from? This is "just" a sample, so it doesn't need to be perfect by any means, but just curious as why it's 4 x the MTU?

Reply:
It's trial and error.
I started from 8x , and reduced step by step.
When otc_obj_buf is less than 4x , it always ASSERT in net_buf_simple_tailroom  .

Reply:
But why? 

If it fails below 4, then it's clearly due to something, and it'd be much better if we knew what that was, so that we could use that. If it's due to some other value (e.g. `BT_L2CAP_TX_BUF_COUNT`) then we should use that value instead of the magic number 4.

Reply:
Use static buffer instead of net_buf.